webapp/src/index.tsx

```diff
@@ -21,9 +21,7 @@ function getSettings(serverRoute: string, disabled: boolean) {
                 title: 'Connect your Microsoft Teams Account',
                 text: 'Connect your Mattermost and Microsoft Teams accounts to get the ability to link and synchronise channel-based collaboration with Microsoft Teams.',
                 buttonText: 'Connect account',
-                onClick: () => Client.connect().then((result) => {
-                    window.open(result?.connectUrl, '_blank');
-                }),
+                onClick: () => Client.connect(),
```

Comment:
I think this still needs to be in a new window

```suggestion
                onClick: () => window.open(`${Client.url}/connect`),
```

(...or with a modified client method)
```suggestion
            onClick: () => window.open(Client.getConnectRoute()),
```